test/test-db.js

```diff
@@ -174,7 +174,57 @@ describe("mongoose-field-encryption plugin db", function() {
       });
   });
 
-  it("should not encrypt non string fields on update", () => {
+  it("should encrypt string fields on update without $set", () => {
+    // given
+    let sut = getSut();
+
+    // when
+    return sut
+      .save()
+      .then(() => {
+        expectEncryptionValues(sut);
+
+        return NestedFieldEncryption.update(
+          { _id: sut._id },
+          { toEncryptString: "snoop" }
+        );
+      })
+      .then(() => {
+        return NestedFieldEncryption.findById(sut._id);
+      })
+      .then(found => {
+        // then
+        expect(found.__enc_toEncryptString).to.be.false;
+        expect(found.toEncryptString).to.equal("snoop");
+      });
+  });
+
+  it("should encrypt string fields on fineOneAndUpdate", () => {
+    // given
+    let sut = getSut();
+
+    // when
+    return sut
+      .save()
+      .then(() => {
+        expectEncryptionValues(sut);
+
+        return NestedFieldEncryption.findOneAndUpdate(
+          { _id: sut._id },
+          { $set: { toEncryptString: "snoop", __enc_toEncryptString: false } }
+        );
+      })
+      .then(() => {
+        return NestedFieldEncryption.findById(sut._id);
+      })
+      .then(found => {
+        // then
+        expect(found.__enc_toEncryptString).to.be.false;
+        expect(found.toEncryptString).to.equal("snoop");
+      });
+  })
+
+  it("should encrypt non string fields on update", () => {
```

Comment:
This test here needs to be fixed. The error that is caught is actually the failure condition. This is actually my fault as I did not write it thoroughly enough :)

Instead of the following:
```
      .then(() => {
        expect.fail("should not have updated");
      })
      .catch(err => {
        // then
        // TODO: this is a mongoose cast error
        expect(err).to.not.be.null;
      });
```

Need to test that the operation was successful!
```
      .then(() => {
        return NestedFieldEncryption.findById(sut._id);
      })
      .then(found => {
        // TODO: add expectations here
      });
```